wasm/src/main/java/com/dylibso/chicory/wasm/Encoding.java

```diff
@@ -11,6 +13,126 @@ public final class Encoding {
+     *
+     * @param buffer the byte buffer
+     * @return the resulting long
+     */
+    public static long readVarUInt32(ByteBuffer buffer) {
+        var value = readUnsignedLeb128(buffer, MAX_VARINT_LEN_32);
+        if (value < 0 || value > MAX_UNSIGNED_INT) {
+            throw new MalformedException("integer too large");
+        }
+        return value;
+    }
+
+    /**
+     * Read a signed I32 from the buffer. We can't fit an unsigned 32bit int into a java int, so we must use a long to use the same type as unsigned.
+     * See <a href="https://www.w3.org/TR/wasm-core-1/#integers">2.2.2. Integers</a> of the WebAssembly Core Specification.
+     *
+     * @param buffer the byte buffer
+     * @return the resulting long
+     */
+    public static long readVarSInt32(ByteBuffer buffer) {
+        var value = readSigned32Leb128(buffer);
+        if (value < MIN_SIGNED_INT || value > MAX_SIGNED_INT) {
+            throw new MalformedException("integer too large");
+        }
+        return value;
+    }
+
+    /**
+     * Read a signed I64 from the buffer which fits neatly into a long.
+     * See <a href="https://www.w3.org/TR/wasm-core-1/#integers">2.2.2. Integers</a> of the WebAssembly Core Specification.
+     *
+     * @param buffer the byte buffer
+     * @return the resulting long
+     */
+    public static long readVarSInt64(ByteBuffer buffer) {
+        return readSigned64Leb128(buffer);
+    }
+
+    /**
+     * Read a F64 from the buffer which fits neatly into a long.
+     * See <a href="https://www.w3.org/TR/wasm-core-1/#floating-point">2.2.3. Floating-Point</a> of the WebAssembly Core Specification.
+     *
+     * @param buffer the byte buffer
+     * @return the resulting long
+     */
+    public static long readFloat64(ByteBuffer buffer) {
+        return buffer.getLong();
+    }
+
+    /**
+     * Read a F32 from the buffer which fits neatly into a long.
+     * See <a href="https://www.w3.org/TR/wasm-core-1/#floating-point">2.2.3. Floating-Point</a> of the WebAssembly Core Specification.
+     *
+     * @param buffer the byte buffer
+     * @return the resulting long
+     */
+    public static long readFloat32(ByteBuffer buffer) {
+        return readInt(buffer);
+    }
+
+    /**
+     * Read a symbol name from the buffer as UTF-8 String.
+     * See <a href="https://www.w3.org/TR/wasm-core-1/#names%E2%91%A0">2.2.4. Names</a> of the WebAssembly Core Specification.
+     *
+     * @param buffer the byte buffer
+     * @return the resulting long
+     */
+    public static String readName(ByteBuffer buffer) {
+        return readName(buffer, true);
+    }
+
+    public static String readName(ByteBuffer buffer, boolean checkMalformed) {
+        var length = (int) readVarUInt32(buffer);
+        byte[] bytes = new byte[length];
+        try {
+            readBytes(buffer, bytes);
+        } catch (BufferUnderflowException e) {
```

Comment:
This catch shouldn't be needed as `readBytes` already checks the length.

Or we could replace the checks in those `read` methods with handling for `BufferUnderflowException`.

Reply:
removed